test/e2e/storage/persistent_volumes.go

```diff
@@ -308,45 +310,105 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
 				framework.SkipUnlessProviderIs("openstack", "gce", "gke", "vsphere", "azure")
 
 				numVols := 4
-				pvcs := []*v1.PersistentVolumeClaim{}
+				ssTester := framework.NewStatefulSetTester(c)
 
-				By("Creating PVCs")
+				By("Creating a StatefulSet pod to initialize data")
+				writeCmd := "true"
 				for i := 0; i < numVols; i++ {
-					pvc = framework.MakePersistentVolumeClaim(framework.PersistentVolumeClaimConfig{}, ns)
-					pvc, err = framework.CreatePVC(c, ns, pvc)
-					Expect(err).NotTo(HaveOccurred())
-					pvcs = append(pvcs, pvc)
+					writeCmd += fmt.Sprintf("&& touch %v", getVolumeFile(i))
 				}
-
-				By("Waiting for PVCs to be bound")
-				for _, pvc := range pvcs {
-					framework.Logf("Created PVC %q", pvc.Name)
-					framework.ExpectNoError(framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, c, ns, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout))
+				writeCmd += "&& sleep 10000"
+
+				probe := &v1.Probe{
+					Handler: v1.Handler{
+						Exec: &v1.ExecAction{
+							// Check that the last file got created
+							Command: []string{"test", "-f", getVolumeFile(numVols - 1)},
+						},
+					},
+					InitialDelaySeconds: 1,
+					PeriodSeconds:       1,
 				}
 
-				By("Creating a pod and initializing data")
-				writeCmd := "true"
-				for i, pvc := range pvcs {
-					// mountPath is /mnt/volume<i+1>
-					writeCmd += fmt.Sprintf("&& touch /mnt/volume%v/%v", i+1, pvc.Name)
-				}
-				pod := framework.MakePod(ns, nil, pvcs, false, writeCmd)
-				pod, err = c.CoreV1().Pods(ns).Create(pod)
+				spec := makeStatefulSetWithPVCs(ns, writeCmd, numVols, probe)
```

Comment:
from the top level of this test it is not clear where the volume mount names are coming from, it would be better if you could somehow extract the volume mount paths out to this top level and pass it into this function so that the test is slightly more readable.
#testingOnTheToiletEp529